cookbooks/cdo-varnish/test/shared/shared.rb

```diff
@@ -13,391 +14,423 @@
 require 'securerandom'
 require 'uri/common'
 
-# Chef node attributes as constants
-DASHBOARD_PORT = 8080
-PEGASUS_PORT = 8081
-VARNISH_PORT = 80
-LOCALHOST = 'localhost'
-VARNISH_URL = "#{LOCALHOST}:#{VARNISH_PORT}"
-ENVIRONMENT = 'integration'
-CLOUDFRONT_URL = "https://#{ENVIRONMENT}-studio.code.org"
-
 puts "Cloudfront: #{$cloudfront}"
 # Setup the mock-server and ngrok
-puts 'setup'
-NGROK_PID = $cloudfront && spawn('/usr/local/ngrok/ngrok start cdo --config=/home/kitchen/.ngrok2/ngrok.yml --log=stdout')
-PID = spawn('cd ~; java -jar mock.jar --verbose')
-# Don't start tests until wiremock is live.
-mock_started = false
-until mock_started
-  `sleep 1`
-  puts 'testing mock'
-  `curl -s -i #{LOCALHOST}:#{DASHBOARD_PORT}/start`
-  mock_started = $?.exitstatus == 0
-end
-puts 'setup finished'
-
-def teardown
-  puts 'teardown'
-  `curl -s -X POST #{LOCALHOST}:#{DASHBOARD_PORT}/__admin/shutdown`
-  Process.wait PID
-  if $cloudfront
-    `kill -INT #{NGROK_PID}`
-    Process.wait NGROK_PID
-  end
-  puts 'teardown finished'
-end
-
-describe 'varnish' do
-  it 'is installed' do
-    assert_equal '/usr/sbin/varnishd', `which varnishd`.strip
-  end
-
-  it 'is running' do
-    assert_equal 0, (`service varnish status` && $?.exitstatus)
-  end
+module Cdo
+  module MockServer
+    # Chef node attributes as constants
+    DASHBOARD_PORT = 8080
+    PEGASUS_PORT = 8081
+    VARNISH_PORT = 80
+    LOCALHOST = 'localhost'
+    VARNISH_URL = "#{LOCALHOST}:#{VARNISH_PORT}"
+    CLOUDFRONT_URL = "https://#{$environment}-studio.code.org"
+
+    def self.setup
+      puts 'setup'
+      $ngrok_pid = $cloudfront && spawn('/usr/local/ngrok/ngrok start cdo --config=/home/kitchen/.ngrok2/ngrok.yml --log=stdout')
+      $pid = spawn('cd ~; java -jar mock.jar --verbose')
```

Comment:
Java will always be installed here, since this code is run within the managed test-kitchen container that's explicitly provisioned with Java package.